api/src/test/java/org/pmiops/workbench/api/ClusterControllerTest.java

```diff
@@ -418,6 +424,31 @@ public void testLocalize_noNotebooks() throws Exception {
     assertThat(resp.getClusterLocalDirectory()).isEqualTo("workspaces/wsid");
   }
 
+  @Test(expected = ForbiddenException.class)
+  public void listCluster_requireActiveBilling() {
+    testWorkspace.setBillingStatus(BillingStatus.INACTIVE);
+
+    doThrow(ForbiddenException.class)
+        .when(workspaceService)
+        .requireActiveBilling(WORKSPACE_NS, WORKSPACE_ID);
+
+    clusterController.listClusters(WORKSPACE_NS, WORKSPACE_ID);
+  }
+
+  @Test(expected = ForbiddenException.class)
+  public void localize_requireActiveBilling() {
+    testWorkspace.setBillingStatus(BillingStatus.INACTIVE);
+
+    doThrow(ForbiddenException.class)
+        .when(workspaceService)
+        .requireActiveBilling(WORKSPACE_NS, WORKSPACE_ID);
```

Comment:
So the only assertion is that the test throws this exception, and you're only throwing the exception because of Mockito. This feels like you're possibly testing this behavior at the wrong level, or in the wrong class.

Reply:
I was thinking about this a bit. I think just verifying that `requireActiveBilling()` is called is enough here, since we test that it throws or not correctly elsewhere. That way you could drop the `expected` and the `doThrow` in favor of one statement that's a bit clearer I think.